tempest/api/workloadmgr/base.py

```diff
@@ -3723,3 +3723,106 @@ def verify_network_restore(self, nt_bf, nt_af, sbnt_bf, sbnt_af, rt_bf,
         except Exception as e:
             LOG.debug("Exception in verify_network_restore: " + str(e))
 
+    '''
+    download test image
+    '''
+
+    def download_image(self,
+            url=tvaultconf.image_url,
+            filename=tvaultconf.image_filename):
+        try:
+            f = open(filename,'wb')
+            f.write(requests.get(url+"/"+filename).content)
```

Comment:
If any exception occurs during f.write then f.close() will not get execute. Better to use "with open(filename, 'wb') as f"